src/App.tsx

```diff
@@ -17,18 +56,32 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onFilterChange={handleFilterChange}
+                onQueryChange={handleQueryChange}
+                query={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading
+                ? <Loader />
+                : (
+                  <TodoList
+                    todos={visibleTodos}
+                    onChoose={handleTodoSelect}
```

Comment:
Look through your naming, use descriptive names
```suggestion
                    onTodoSelect={handleTodoSelect}
```